### ISSUE TITLE
Referencing module outputs that don't exist doesn't raise an error

### ISSUE THREAD

**Issue by bitglue** [hashicorp/terraform#1448]
Say I have a module `a` with no outputs, and another module `b` which requires a variable `foo`. Then this configuration:

```
module "a" {
    source = "a"
    # this module has no outputs
}

module "b" {
    source = "b"
    foo = "${module.a.this_output_does_not_exist}"
}
```

seems to run just fine. Of course `${module.a.this_output_does_not_exist}` is undefined, so there's no way this could do anything useful, and it should raise an error.

**Comment by bitglue** [comment]
Complete test case at https://github.com/bitglue/terraform-1448

**Comment by justincampbell** [comment]
This bug I ran into seems similar https://gist.github.com/justincampbell/2a32edb75cc2ae16d0d5

**Comment by mitchellh** [comment]
Fixed in #1744. Looks like we were validating this, but just not in the specific case that outputs were used as an input to modules. Test added in PR and fix.
